test/rhsmlib/dbus/test_register.py

```diff
@@ -321,257 +312,162 @@
-        self.interface = dbus.Interface(self.proxy, constants.REGISTER_INTERFACE)
-
-        self.mock_identity.is_valid.return_value = True
-
-        self.mock_cp_provider = mock.Mock(spec=CPProvider, name="CPProvider")
-        self.mock_cp = mock.Mock(spec=connection.UEPConnection, name="UEPConnection")
-
-        # Mock a basic auth connection
-        self.mock_cp.username = "username"
-        self.mock_cp.password = "password"
-
-        self.mock_cp.getOwnerList = mock.Mock()
-        self.mock_cp.getOwnerList.return_value = json.loads(OWNERS_CONTENT_JSON)
-
-        # For the tests in which it's used, the consumer_auth cp and basic_auth cp can be the same
-        self.mock_cp_provider.get_consumer_auth_cp.return_value = self.mock_cp
-        self.mock_cp_provider.get_basic_auth_cp.return_value = self.mock_cp
-
-        register_patcher = mock.patch("rhsmlib.dbus.objects.register.RegisterService", autospec=True)
-        self.mock_register = register_patcher.start().return_value
-        self.addCleanup(register_patcher.stop)
-
-        cert_invoker_patcher = mock.patch("rhsmlib.dbus.objects.register.EntCertActionInvoker", autospec=True)
-        self.mock_cert_invoker = cert_invoker_patcher.start().return_value
-        self.addCleanup(cert_invoker_patcher.stop)
-
-        attach_patcher = mock.patch("rhsmlib.dbus.objects.register.AttachService")
-        self.mock_attach_invoker = attach_patcher.start()
-        self.addCleanup(attach_patcher.stop)
-
-        ent_cert_service_patcher = mock.patch("rhsmlib.dbus.objects.register.EntitlementService")
-        self.mock_ent_cert_service_invoker = ent_cert_service_patcher.start()
-        self.addCleanup(ent_cert_service_patcher.stop)
-
-    def injection_definitions(self, *args, **kwargs):
-        if args[0] == inj.IDENTITY:
-            return self.mock_identity
-        elif args[0] == inj.CP_PROVIDER:
-            return self.mock_cp_provider
-        else:
-            return None
-
-    def test_open_domain_socket(self):
-        dbus_method_args = [""]
-
-        def assertions(*args):
-            result = args[0]
-            self.assertRegex(result, r"/run/dbus.*")
-
-        self.dbus_request(assertions, self.interface.Start, dbus_method_args)
-
-    def test_same_socket_on_subsequent_opens(self):
-        dbus_method_args = [""]
-
-        def assertions(*args):
-            # Assign the result as an attribute to this function.
-            # See http://stackoverflow.com/a/27910553/6124862
-            assertions.result = args[0]
-            self.assertRegex(assertions.result, r"/run/dbus.*")
-
-        self.dbus_request(assertions, self.interface.Start, dbus_method_args)
+class RegisterDBusObjectTest(DBusServerStubProvider):
+    dbus_class = RegisterDBusObject
+    dbus_class_kwargs = {}
 
-        # Reset the handler_complete_event so we'll block for the second
-        # dbus_request
-        self.handler_complete_event.clear()
+    def tearDown(self) -> None:
```

Comment:
this needs to call `tearDown()` of the parent class (`DBusServerStubProvider`), so it is used as expected